samples/snippets/sample_custom_types.py

```diff
@@ -0,0 +1,525 @@
+    request.instance_resource = instance
+
+    # Wait for the create operation to complete.
+    print(
+        f"Creating the {instance_name} instance of type {instance.machine_type} in {zone}..."
+    )
+    operation = instance_client.insert(request=request)
+    while operation.status != compute_v1.Operation.Status.DONE:
+        operation = operation_client.wait(
+            operation=operation.name, zone=zone, project=project_id
+        )
+    if operation.error:
+        print("Error during creation:", operation.error, file=sys.stderr)
+    if operation.warnings:
+        print("Warning during creation:", operation.warnings, file=sys.stderr)
+    print(f"Instance {instance_name} created.")
+    return instance_client.get(project=project_id, zone=zone, instance=instance.name)
+
+
+# [END compute_custom_machine_type_create ]
+
+
+# [START compute_custom_machine_type_create_with_helper ]
+def create_custom_machine(
+    project_id: str,
+    zone: str,
+    instance_name: str,
+    cpu_series: CustomMachineType.CPUSeries,
+    core_count: int,
+    memory: int,
+) -> compute_v1.Instance:
+    """
+    Create a new VM instance with a custom type.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone: name of the zone to create the instance in. For example: "us-west3-b"
+        instance_name: name of the new virtual machine (VM) instance.
+        cpu_series: The type of CPU you want to use. Pick one value from the CustomMachineType.CPUSeries enum.
+            For example: CustomMachineType.CPUSeries.N2
+        core_count: Number of CPU cores you want to use.
+        memory: Number of megabytes of memory you want to use.
+
+    Return:
+        Instance object.
+    """
+    assert cpu_series in (
+        CustomMachineType.CPUSeries.E2,
+        CustomMachineType.CPUSeries.N1,
+        CustomMachineType.CPUSeries.N2,
+        CustomMachineType.CPUSeries.N2D,
+    )
+    custom_type = CustomMachineType(zone, cpu_series, memory, core_count)
+    return create_instance(project_id, zone, instance_name, custom_type)
+
+
+# [END compute_custom_machine_type_create_with_helper ]
+
+
+# [START compute_custom_machine_type_create_shared_with_helper ]
+def create_custom_shared_core_machine(
+    project_id: str,
+    zone: str,
+    instance_name: str,
+    cpu_series: CustomMachineType.CPUSeries,
+    memory: int,
+):
+    """
+    Create a new VM instance with a custom type using shared CPUs.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you want to use.
+        zone: name of the zone to create the instance in. For example: "us-west3-b"
+        instance_name: name of the new virtual machine (VM) instance.
+        cpu_series: The type of CPU you want to use. Pick one value from the CustomMachineType.CPUSeries enum.
+            For example: CustomMachineType.CPUSeries.E2_MICRO
+        memory: Number of megabytes of memory you want to use.
```

Comment:
memory: the amount of memory for the VM instance, in megabytes.
Please change this globally.
It'll be good to provide an example here. For example, 3072MB. Is this correct?